simulator/syncer/syncer.go

```diff
@@ -5,72 +5,61 @@ import (
 
 	"golang.org/x/xerrors"
 	"k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/dynamic/dynamicinformer"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
-)
 
-type Service struct {
-	clients *Clients
+	"sigs.k8s.io/kube-scheduler-simulator/simulator/resourceapplier"
+)
 
-	gvrs               []schema.GroupVersionResource
-	mutatingFunctions  map[schema.GroupVersionResource][]MutatingFunction
-	filteringFunctions map[schema.GroupVersionResource][]FilteringFunction
+// DefaultGVRs is a list of GroupVersionResource that we sync by default (configurable with Options),
+// which is a suitable resource set for the vanilla scheduler.
+//
+// Note that this order matters - When first importing resources, we want to sync namespaces first, then priorityclasses, storageclasses...
+var DefaultGVRs = []schema.GroupVersionResource{
+	{Group: "", Version: "v1", Resource: "namespaces"},
+	{Group: "scheduling.k8s.io", Version: "v1", Resource: "priorityclasses"},
+	{Group: "storage.k8s.io", Version: "v1", Resource: "storageclasses"},
+	{Group: "", Version: "v1", Resource: "persistentvolumeclaims"},
+	{Group: "", Version: "v1", Resource: "nodes"},
+	{Group: "", Version: "v1", Resource: "persistentvolumes"},
+	{Group: "", Version: "v1", Resource: "pods"},
 }
 
-// Note: Clients and its fields are exposed intentionally so that users can use it in MutatingFunction and FilteringFunction.
-type Clients struct {
-	// SrcDynamicClient is the dynamic client for the source cluster, which the resource is supposed to be copied from.
-	SrcDynamicClient dynamic.Interface
-	// DestDynamicClient is the dynamic client for the destination cluster, which the resource is supposed to be copied to.
-	DestDynamicClient dynamic.Interface
-	RestMapper        meta.RESTMapper
+type Service struct {
+	gvrs                   []schema.GroupVersionResource
+	srcDynamicClient       dynamic.Interface
+	resourceApplierService *resourceapplier.Service
 }
 
 type Options struct {
 	// GVRsToSync is a list of GroupVersionResource that will be synced.
 	// If GVRsToSync is nil, defaultGVRs are used.
 	GVRsToSync []schema.GroupVersionResource
```

Comment:
Can we somehow move it to the resourceapplier option as well so that we can unify the interface of the option regardless of whether users want to use oneshot or syncer?

Reply:
We could just make `resourceapplier` have `GVRsToSync` and `syncer` refer to `resouceApplierService.GRVsToSync`.

```go
	// in syncer service
	infFact := dynamicinformer.NewFilteredDynamicSharedInformerFactory(s.srcDynamicClient, 0, metav1.NamespaceAll, nil)
	for _, gvr := range s.resourceApplierService.GVRsToSync { // syncer itself doesn't have gvrs
		inf := infFact.ForResource(gvr).Informer()
```

`oneshotimporter` would just ignore it.

Reply:
> oneshotimporter would just ignore it.

Well, can't we make oneshotimporter also configurable about which resource to import with the same option of resourceapplyer?

Reply:
Yes. So we could make a PR that makes it configurable after this PR.

For now, I'll just implement as I wrote in the last comment.

Reply:
Ah, got what you meant. Yes, I'm fine with it if oneshotimporter would follow the option later.

Reply:
I unified the syncer options into the resourceapplier options.

Reply:
And also update the doc.